automation_tools/repository.py

```diff
@@ -0,0 +1,126 @@
+    it will delete de repository file.
+
+    """
+    run(
+        'subscription-manager repos --disable {0}'
+        .format(' '.join(['"{0}"'.format(repo) for repo in args]))
+    )
+
+    for name, delete in kwargs.items():
+        repo_file = '/etc/yum.repos.d/{0}.repo'.format(name)
+        if delete:
+            run('rm {0}'.format(repo_file))
+        else:
+            run('sed -i -e "s/^enabled=.*/enabled=0/" {0}'.format(repo_file))
+
+
+def enable_repos(*args, **kwargs):
+    """Enable repos passed as ``args`` and create one repofile with each
+    ``kwargs``.
+
+    For example::
+
+        manage_repos('repo1', custom_repo='http://repourl.domain.com/path')
+
+    The above call will result in enabling ``repo1`` using
+    ``subscription-manager repos --enable``. Also will create a
+    custom_repo.repo file with the following contents::
+
+        [custom_repo]
+        name=custom_repo
+        baseurl=http://repourl.domain.com/path
+        enabled=1
+        gpgcheck=0
+
+    """
+    run('subscription-manager repos --enable {0}'
+        .format(' '.join(['"{0}"'.format(repo) for repo in args])))
+
+    for name, url in kwargs.items():
+        repo_file = StringIO()
+        repo_file.write(
+            '[{name}]\n'
+            'name={name}\n'
+            'baseurl={url}\n'
+            'enabled=1\n'
+            'gpgcheck=0\n'
+            .format(name=name, url=url)
+        )
+        put(local_path=repo_file,
+            remote_path='/etc/yum.repos.d/{0}.repo'.format(name))
+        repo_file.close()
+
+
+def enable_satellite_repos(cdn=False, disable_enabled=True):
+    """Enable repositories required to install Satellite 6
+
+    If ``disable_enabled`` is ``True``, then this task will first disable any
+    already enabled repository, including beaker ones.
+
+    """
+    if isinstance(cdn, str):
+        cdn = (cdn.lower() == 'true')
+    if isinstance(disable_enabled, str):
+        disable_enabled = (disable_enabled.lower() == 'true')
+
+    if disable_enabled:
+        disable_beaker_repos()
+        disable_repos('*')
+
+    # Disable yum plugin for sub-man
+    run('sed -i -e "s/^enabled.*/enabled=0/" '
+        '/etc/yum/pluginconf.d/subscription-manager.conf')
+
+    repos = [
+        'rhel-{0}-server-rpms',
+        'rhel-server-rhscl-{0}-rpms',
+    ]
+
+    if cdn is True:
+        repos.append('rhel-{0}-server-satellite-6.0-rpms')
+
+    enable_repos(*[repo.format(distro_info()[1]) for repo in repos])
+
+    run('yum repolist')
+
+
+def disable_beaker_repos():
+    """Disable beaker repositories
+
+    If yum-config-manager is available this task will disable the repos, if not
+    it will move the beaker repo files to the running user home directory
+
+    """
+    # Clean up system if Beaker-based
+    result = run('which yum-config-manager', warn_only=True)
+    if result.succeeded:
+        run('yum-config-manager --disable "beaker*"')
```

Comment:
Neat. This command succeeds even when there are no beaker repositories on the system.

```
# yum-config-manager | grep -i beaker
# ls /etc/yum.repos.d/beaker*
ls: cannot access /etc/yum.repos.d/beaker*: No such file or directory
# yum-config-manager --disable "beaker*"; echo $?
Loaded plugins: product-id
0
```